crates/chisel/bin/main.rs

```diff
@@ -228,3 +228,48 @@ impl Provider for ChiselParser {
         Ok(Map::from([(Config::selected_profile(), Dict::default())]))
     }
 }
+
+/// Evaluate a single Solidity line.
+async fn dispatch_repl_line(dispatcher: &mut ChiselDispatcher, line: &str) {
+    match dispatcher.dispatch(line).await {
+        DispatchResult::Success(msg) | DispatchResult::CommandSuccess(msg) => if let Some(msg) = msg {
+            println!("{}", Paint::green(msg));
+        },
+        DispatchResult::UnrecognizedCommand(e) => eprintln!("{e}"),
+        DispatchResult::SolangParserFailed(e) => {
+            eprintln!("{}", Paint::red("Compilation error"));
+            eprintln!("{}", Paint::red(format!("{e:?}")));
+        }
+        DispatchResult::FileIoError(e) => eprintln!("{}", Paint::red(format!("⚒️ Chisel File IO Error - {e}"))),
+        DispatchResult::CommandFailed(msg) | DispatchResult::Failure(Some(msg)) => eprintln!("{}", Paint::red(msg)),
+        DispatchResult::Failure(None) => eprintln!("{}\nPlease Report this bug as a github issue if it persists: https://github.com/foundry-rs/foundry/issues/new/choose", Paint::red("⚒️ Unknown Chisel Error ⚒️")),
+    }
+}
+
+/// Evaluate multiple Solidity source files contained within a
+/// Chisel prelude directory.
+async fn evaluate_prelude(
+    dispatcher: &mut ChiselDispatcher,
+    maybe_prelude: Option<PathBuf>,
+) -> eyre::Result<()> {
+    if let Some(prelude_dir) = maybe_prelude {
+        let prelude_sources = fs::files_with_ext(prelude_dir, "sol");
```

Comment:
Hmm, might make things just a tad more complicated, but wdyt about supporting both:
- Loading a single file
- Pointing to a directory of files

If you find it blows up the complexity of the command too much, feel free to ignore—just think a lot of ppl would appreciate just putting a file somewhere and loading it

Reply:
I originally just had a single source file supported. @mattsse asked to support multiple files, which I liked... I thought of supporting both files and directories, but it's a bit trickier. I'll see if I find the time to work on that tomorrow.

for my particular use case, a single source file is actually enough. it's possible append multiple `import` statements to a single source file, but then you lose the dynamic nature of not needing to specify which files to import code from

Reply:
no worries @sug0 — did not mean to conflict with mattsse at all! just would love to support both if possible. happy to help getting this over the line if it becomes too much work :)